assignment3/Assignment3.java

```diff
@@ -0,0 +1,53 @@
+package assignment3;
+
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class Assignment3 {
+
+    @Test
+    public void testEmptyDict() {
+        char[][] grid = new char[][] {
+            {'a', 'a', 'r'},
+            {'t', 'c', 'd'}
+        };
+        WordDictionary dict = new WordDictionary();
+        Set<String> expected = new HashSet<>();
+        Set<String> words = new GridWordSearcher(grid.length, grid[0].length, grid, dict).findWords();
+        assertEquals("Set of words is not empty", words, expected);

```

Comment:
Good use of JUnit. For your information, there is a nice library called Truth which we use at Google with JUnit tests, and it is open sourced. https://github.com/google/truth

It lets you write things like assertThat(words).isEmpty() instead of using assertEquals like you have here, and it can make your tests a bit easier to read.